cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -55,31 +41,133 @@ def has_nodes(self):
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):
-        credentials_key = prov_config['credentials']
-        credentials = cls.process_credential_yaml_key(credentials_key)
-        credential_dict = {'default': credentials}
+        endpoints = [RHOSEndpoint(**prov_config['endpoints']['default']),
```

Comment:
**QUESTION:** Should this guard in case all the credentials are not there? And should there be a "type" that sets the class with some dict doing string_type->class mapping on the provider itself? Or is that over complicating it? 
**REQUIRED:** I still think we should guard against a missing credential.